### ISSUE TITLE
[bug] duplicate image name can cause overwrite problem

### ISSUE THREAD

**Issue by linrongc** [Tencent/tencent-ml-images#32]
In the download script, saving images with path "save_dir + im_name" will overwrite any images with same name.
 
For example:
http://i.ytimg.com/vi/6rMwgpPSJyU/3.jpg 8486:1  8479:1  8473:1  5175:1  5170:1  1042:1  865:1   2:1

http://floo.ru/Fotogaleri2/OdontoglossumFoto/3.jpg      5415:1  5296:1  5240:1  5236:1  5189:1  5181:1  5173:1  5170:1  1042:1  865:1   2:1
they have the same image name.

**Comment by wubaoyuan** [comment]
@linrongc Thanks for this feedback.  We will try to fix it asap.

**Comment by linrongc** [comment]
using line_num as save name with mapping file will be an easy fix.

**Comment by wwfnwg** [comment]
这是一个巨坑

**Comment by wubaoyuan** [comment]
@linrongc @wwfnwg I have updated Line 27,28 of 'download_urls_multithreading.py'. Sorry for this bug, if any other bugs, please let me know. Thanks.

**Comment by wwfnwg** [comment]
> @linrongc @wwfnwg I have updated Line 27,28 of 'download_urls_multithreading.py'. Sorry for this bug, if any other bug, please let me know. Thanks.

还是有问题，有些整个url都是相同的，标签却不相同

**Comment by wubaoyuan** [comment]
@wwfnwg Yes. We also find that there is a very small proportion of repeated urls in the provided url file, which is not completely matched with the train image list in our training. We are updating it. The corrected version will be released soon.

**Comment by Northrend** [comment]
Saving image with its whole url path which is url-safe base64 encoded is a better choice

**Comment by wubaoyuan** [comment]
> > @linrongc @wwfnwg I have updated Line 27,28 of 'download_urls_multithreading.py'. Sorry for this bug, if any other bug, please let me know. Thanks.
> 
> 还是有问题，有些整个url都是相同的，标签却不相同

@wwfnwg Please see the updated README about the repeated URLs.
